pkg/common/interfaces.go

```diff
@@ -36,6 +36,8 @@ type Route interface {
 	// StatusCodeResponses defines a mapping of HTTP Status Codes to the specific response(s).
 	// Multiple responses with the same HTTP Status Code are acceptable.
 	StatusCodeResponses() []StatusCodeResponse
+	// Deprecated marks a route as deprecated
+	Deprecated() bool
```

Comment:
Will this be released as a breaking change? iirc, we were quite careful when adding this interface to not break backwards compatibility, suggest doing the same here

Reply:
I did a quick scan of the k/k codebase, it looks like nothing is using the particular interface and all calls of the interface are from kube-openapi itself. The GetOperationIDAndTagsFromRoute in the PR linked has not yet been integrated into k/k yet. 

So to answer your question, yes this will be a breaking change for current users of the API, but at the moment it seems like we have no users of the particular API (that we know of). I'm okay with taking the safe approach of just creating another interface, just want to gather some thoughts first.

Reply:
Thanks for your reply. I agree it isn't a huge breaking change and won't affect k/k, though will affect other projects using it (mine 😄), perhaps it's acceptable for this project to only support k/k.
Do you think it would be valuable to switch k/k to the interface, or rather wait until there's a push to get off go-restful?

Reply:
Thanks for pointing that out, that's enough of a reason to create backwards compatibility and create a new interface. Do you have any preferences for the interface name?

I don't seem any harm in switching k/k to the interface, although that will be a decision up to the approvers.

Reply:
Thanks very much for the backwards compat effort :) No real preference on name, perhaps `DeprecatableRoute`?